js/gestionPresupuesto.js

```diff
@@ -1,25 +1,149 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+// valor = prompt("introduzca un valor");
 
 // TODO: Variable global
+let presupuesto = 0;
+let gastos =[];
+let idGasto = 0;
 
-
-function actualizarPresupuesto() {
+function actualizarPresupuesto(valor) {
     // TODO
+    
+    if ((isNaN(valor))||(valor<0)){
+        return -1;
+    }else{
+        presupuesto = valor;
+    }
+    return presupuesto;
 }
+        
+    
 
 function mostrarPresupuesto() {
     // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`;
+     
+    
 }
 
-function CrearGasto() {
+function CrearGasto(descripcion,valor,fecha,...etiquetas) {
     // TODO
-}
+    this.descripcion = descripcion;
+        
+        if ((isNaN(valor))||(valor<0)){
+            this.valor =0;
+        }else{
+            this.valor = valor;
+        }
+
+        if(!fecha){
+            this.fecha = new Date().toISOString();
+        }else{
+            this.fecha = Date.parse(fecha);
+        }
+
+        this.etiquetas = etiquetas.length ? etiquetas : [];
```

Comment:
Puedes utilizar también `this.anyadirEtiqueta` para evitar crear un gasto con etiquetas duplicadas.